packages/playground/src/weblets/tf_deployment_list.vue

```diff
@@ -476,53 +486,112 @@ const deletingDialog = ref(false);
 const table = ref() as Ref<{ loadDeployments(): void }>;
 const gridStore = useGrid();
 const grid = gridStore.client as GridClient;
+const hasWorkers = computed(() => selectedItems.value.map(item => item.workers && item.workers.length).some(i => i));
 
 const _idx = tabs.findIndex(t => t.value === props.projectName);
 const activeTab = ref(!props.projectName ? 0 : _idx) as Ref<number>;
 watch(activeTab, () => (selectedItems.value = []));
 
 async function onDelete(k8s = false) {
+  try {
+    prepareForDeletion();
+    await processSelectedItems(k8s);
+    refreshDeployments();
+  } catch (e) {
+    handleError(e);
+  } finally {
+    finalizeDeletion();
+  }
+}
+
+function prepareForDeletion() {
   deletingDialog.value = false;
   deleting.value = true;
+}
+
+async function processSelectedItems(k8s: boolean) {
+  const itemsToDelete = getAllItemsToDelete();
+
+  for (const item of itemsToDelete) {
+    await deleteItem(item, k8s);
+  }
+}
+
+function getAllItemsToDelete() {
+  const itemsToDelete = [...selectedItems.value];
+
+  selectedItems.value.forEach(item => {
+    if (item?.projectName.toLowerCase().includes(ProjectName.Caprover.toLowerCase())) {
+      itemsToDelete.push(...item.workers);
+    }
+  });
+
+  return itemsToDelete;
+}
+
+async function deleteItem(item: any, k8s: boolean) {
   try {
-    for (const item of selectedItems.value) {
-      try {
-        if (props.projectName?.toLowerCase() === ProjectName.Domains.toLowerCase()) {
-          await deleteGatewayDeployment(
-            updateGrid(grid, { projectName: props.projectName.toLocaleLowerCase() }),
-            item[0].workloads[0].name as string,
-          );
-        } else {
-          await deleteDeployment(updateGrid(grid!, { projectName: item.projectName }), {
-            deploymentName: item.deploymentName,
-            name: k8s ? item.deploymentName : item.name,
-            projectName: item.projectName,
-            ip: item.interfaces?.[0]?.ip,
-            k8s,
-          });
-        }
-      } catch (e: any) {
-        createCustomToast(`Failed to delete deployment with name: ${item.name}`, ToastType.danger);
-        console.log("Error while deleting deployment", e.message);
-        continue;
-      }
+    if (isDomainProject()) {
+      await deleteDomainProject(item);
+    } else {
+      await deleteStandardItem(item, k8s);
     }
-    table.value?.loadDeployments();
-  } catch (e) {
-    createCustomToast((e as Error).message, ToastType.danger);
-  } finally {
-    selectedItems.value = [];
-    deleting.value = false;
+  } catch (e: any) {
+    logDeletionError(item.name, e);
   }
 }
```

Comment:
Do we need to seperate these into 3 functions?

Reply:
Why not?